e2e/nfs.go

```diff
@@ -0,0 +1,564 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo" // nolint
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+var (
+	nfsProvisioner     = "csi-nfsplugin-provisioner.yaml"
+	nfsProvisionerRBAC = "csi-provisioner-rbac.yaml"
+	nfsProvisionerPSP  = "csi-provisioner-psp.yaml"
+	nfsNodePlugin      = "csi-nfsplugin.yaml"
+	nfsNodePluginRBAC  = "csi-nodeplugin-rbac.yaml"
+	nfsNodePluginPSP   = "csi-nodeplugin-psp.yaml"
+	nfsRookCephNFS     = "rook-nfs.yaml"
+	nfsDeploymentName  = "csi-nfsplugin-provisioner"
+	nfsDeamonSetName   = "csi-nfs-node"
+	nfsDirPath         = "../deploy/nfs/kubernetes/"
+	nfsExamplePath     = examplePath + "nfs/"
+	nfsPoolName        = ".nfs"
+
+	// FIXME: some tests change the subvolumegroup to "e2e".
+	defaultSubvolumegroup = "csi"
+)
+
+func deployNFSPlugin(f *framework.Framework) {
+	// delete objects deployed by rook
+
+	err := deleteResource(nfsDirPath + nfsProvisionerRBAC)
+	if err != nil {
+		e2elog.Failf("failed to delete provisioner rbac %s: %v", nfsDirPath+nfsProvisionerRBAC, err)
+	}
+
+	err = deleteResource(nfsDirPath + nfsNodePluginRBAC)
+	if err != nil {
+		e2elog.Failf("failed to delete nodeplugin rbac %s: %v", nfsDirPath+nfsNodePluginRBAC, err)
+	}
+
+	// the pool should not be deleted, as it may contain configurations
+	// from non-e2e related CephNFS objects
+	err = createPool(f, nfsPoolName)
```

Comment:
as we are creating the test pool, we can create and delete it as part of testing?

Reply:
in case the pool already exists, it may have contents from configurations outside of this e2e test. In that case, deleting the pool should not be done